src/DynamoCore/Extensions/LinterExtensionBase.cs

```diff
@@ -71,21 +73,41 @@ public void RemoveLinterRule(LinterRule linterRule)
         /// </summary>
         internal void Activate()
         {
-            if (IsActive)
+            //Nothing left to do if initial setup is complete and we are already linked to application events
+            if (SetupComplete)
```

Comment:
I have introduced this new flag so that you can't activate it multiple times.
In theory, one can keep the same Linter active and play with Activate/Deactivate through the API.